src/main/java/org/jabref/gui/importer/fetcher/WebSearchPane.java

```diff
@@ -0,0 +1,119 @@
+import org.jabref.gui.help.HelpAction;
+import org.jabref.gui.icon.IconTheme;
+import org.jabref.gui.search.SearchTextField;
+import org.jabref.gui.util.ViewModelListCellFactory;
+import org.jabref.logic.importer.SearchBasedFetcher;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.preferences.JabRefPreferences;
+
+import org.fxmisc.easybind.EasyBind;
+
+public class WebSearchPane extends SidePaneComponent {
+
+    private final JabRefFrame frame;
+    private final JabRefPreferences preferences;
+    private final WebSearchPaneViewModel viewModel;
+
+    public WebSearchPane(SidePaneManager sidePaneManager, JabRefPreferences preferences, JabRefFrame frame) {
+        super(sidePaneManager, IconTheme.JabRefIcons.WWW, Localization.lang("Web search"));
+        this.frame = frame;
+        this.preferences = preferences;
+        this.viewModel = new WebSearchPaneViewModel(preferences.getImportFormatPreferences(), frame);
+    }
+
+    @Override
+    public Action getToggleAction() {
+        return StandardActions.TOGGLE_WEB_SEARCH;
+    }
+
+    @Override
+    protected Node createContentPane() {
+        // Setup combo box for fetchers
+        ComboBox<SearchBasedFetcher> fetchers = new ComboBox<>();
+        new ViewModelListCellFactory<SearchBasedFetcher>()
+                .withText(SearchBasedFetcher::getName)
+                .install(fetchers);
+        viewModel.selectedFetcherProperty().bind(fetchers.getSelectionModel().selectedItemProperty());
+        fetchers.itemsProperty().bind(viewModel.fetchersProperty());
+        fetchers.setMaxWidth(Double.POSITIVE_INFINITY);
+
+        // Create help button for currently selected fetcher
+        StackPane helpButtonContainer = new StackPane();
+        ActionFactory factory = new ActionFactory(preferences.getKeyBindingRepository());
+        EasyBind.subscribe(viewModel.selectedFetcherProperty(), fetcher -> {
+            if (fetcher != null && fetcher.getHelpPage().isPresent()) {
+                HelpAction helpCommand = new HelpAction(fetcher.getHelpPage().get());
+                Button helpButton = factory.createIconButton(StandardActions.HELP, helpCommand.getCommand());
+                helpButtonContainer.getChildren().setAll(helpButton);
+            } else {
+                helpButtonContainer.getChildren().clear();
+            }
+        });
+        HBox fetcherContainer = new HBox(fetchers, helpButtonContainer);
+        HBox.setHgrow(fetchers, Priority.ALWAYS);
+
+        // Create text field for query input
+        TextField query = SearchTextField.create();
+        viewModel.queryProperty().bind(query.textProperty());
+
+        // Create button that triggers search
+        Button search = new Button(Localization.lang("Search"));
+        search.setOnAction(event -> viewModel.search());
+
+        // Poor mans binding of default index
+        // (this logic should actually be in the view model but selected item property is readonly)
```

Comment:
I don't understand your comment. 
Selected item property is bidiretional: getSelectionModel().selectedItemProperty()
You only have to make sure that the Observable List is initialized before setting the properties.

Reply:
Thanks for the hint. Works like a charm!